[Bug]: Cannot create QuickSight dataset which uses project operation

### Terraform Core Version

1.4.5

### AWS Provider Version

4.62.0

### Affected Resource(s)

`aws_quicksight_data_set`

### Expected Behavior

Should be able to create QuickSight dataset

### Actual Behavior

Error was returned

### Relevant Error/Panic Output Snippet

```
aws_quicksight_data_set.test: Creating...
╷
│ Error: error creating QuickSight Data Set: InvalidParameter: 1 validation error(s) found.
│ - missing required field, CreateDataSetInput.LogicalTableMap[id].DataTransforms[0].ProjectOperation.ProjectedColumns.
│ 
│ 
│   with aws_quicksight_data_set.test,
│   on main.tf line 58, in resource "aws_quicksight_data_set" "test":
│   58: resource "aws_quicksight_data_set" "test" {
│
```


### Terraform Configuration Files

```terraform
provider "aws" {
  region = "eu-west-1"
}

resource "random_uuid" "val" {}

resource "aws_s3_bucket" "bucket" {
  bucket = random_uuid.val.result
}

resource "aws_s3_object" "object" {
  bucket = aws_s3_bucket.bucket.id
  key    = "manifest.json"
  content = <<EOT
{
    "fileLocations": [ { "URIs": [] } ],
    "globalUploadSettings": { "format": "JSON" }
}
EOT
}

resource "aws_quicksight_data_source" "s3" {
  data_source_id = "example-id"
  name           = "S3"
  type = "S3"

  parameters {
    s3 {
      manifest_file_location {
        bucket = aws_s3_bucket.bucket.id
        key    = "manifest.json"
      }
    }
  }
}

resource "aws_quicksight_data_set" "test" {

  data_set_id    = "test"
  import_mode    = "SPICE"
  name           = "Test"

  physical_table_map {
    physical_table_map_id = "id"
    s3_source {
      data_source_arn = aws_quicksight_data_source.s3.arn
      input_columns {
        name = "var"
        type = "STRING"
      }
      upload_settings {
        format = "JSON"
      }
    }
  }

  logical_table_map {
    logical_table_map_id = "id"
    alias = "alias"
    source {
      physical_table_id = "id"
    }
    data_transforms {
      project_operation {
        projected_columns = [
          "var"
        ]
      }
    }
  }
}
```

### Steps to Reproduce

Plan and apply the supplied Terraform configuration.

### Debug Output

```
aws_quicksight_data_set.test: Creating...
2023-04-13T15:32:08.577+0100 [INFO]  Starting apply for aws_quicksight_data_set.test
2023-04-13T15:32:08.577+0100 [DEBUG] aws_quicksight_data_set.test: applying the planned Create change
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.source.0.join_instruction" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.data_transforms.0.create_columns_operation" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.data_transforms.0.filter_operation" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.data_transforms.0.rename_column_operation" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.data_transforms.0.tag_column_operation" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.data_transforms.0.untag_column_operation" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.data_transforms.0.cast_column_type_operation" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.source.0.join_instruction" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.source.0.join_instruction" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.data_transforms.0.rename_column_operation" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.data_transforms.0.tag_column_operation" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.data_transforms.0.untag_column_operation" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.data_transforms.0.cast_column_type_operation" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.data_transforms.0.create_columns_operation" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.data_transforms.0.filter_operation" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.data_transforms.0.create_columns_operation" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.data_transforms.0.filter_operation" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.data_transforms.0.rename_column_operation" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.data_transforms.0.tag_column_operation" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.data_transforms.0.untag_column_operation" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "logical_table_map.0.data_transforms.0.cast_column_type_operation" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "data_set_usage_configuration" from ComputedKeys
2023-04-13T15:32:08.578+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] setting computed for "tags_all" from ComputedKeys
2023-04-13T15:32:08.579+0100 [ERROR] provider.terraform-provider-aws_v4.62.0_x5: Response contains error diagnostic: @caller=github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/diag/diagnostics.go:55 diagnostic_detail= tf_resource_type=aws_quicksight_data_set tf_proto_version=5.3 tf_provider_addr=registry.terraform.io/hashicorp/aws @module=sdk.proto diagnostic_severity=ERROR diagnostic_summary="error creating QuickSight Data Set: InvalidParameter: 1 validation error(s) found.
- missing required field, CreateDataSetInput.LogicalTableMap[id].DataTransforms[0].ProjectOperation.ProjectedColumns.
" tf_req_id=c3e2dc80-e1b6-99d6-0e07-d49b8ed61664 tf_rpc=ApplyResourceChange timestamp=2023-04-13T15:32:08.579+0100
2023-04-13T15:32:08.602+0100 [ERROR] vertex "aws_quicksight_data_set.test" error: error creating QuickSight Data Set: InvalidParameter: 1 validation error(s) found.
- missing required field, CreateDataSetInput.LogicalTableMap[id].DataTransforms[0].ProjectOperation.ProjectedColumns.
2023-04-13T15:32:08.799+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] Waiting for state to become: [success]
2023-04-13T15:32:08.799+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: HTTP Request Sent: aws.service=S3 http.flavor=1.1 http.request.header.authorization="AWS4-HMAC-SHA256 Credential=ASIA************WFMH/20230413/eu-west-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date;x-amz-security-token, Signature=*****" http.request.header.x_amz_security_token=***** tf_mux_provider=*schema.GRPCProviderServer tf_req_id=09b7b2d1-5bdb-8486-6e08-f3852a8631b3 @caller=github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2@v2.0.0-beta.26/logger.go:90 aws.operation=HeadObject aws.sdk=aws-sdk-go http.user_agent="APN/1.0 HashiCorp/1.0 Terraform/1.4.5 (+https://www.terraform.io) terraform-provider-aws/4.62.0 (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go/1.44.237 (go1.19.7; linux; amd64)" net.peer.name=0c79c024-7ebb-d260-2f8b-8630b1dfc45d.s3.eu-west-1.amazonaws.com aws.region=eu-west-1 http.method=HEAD http.request.body= http.url=https://0c79c024-7ebb-d260-2f8b-8630b1dfc45d.s3.eu-west-1.amazonaws.com/manifest.json tf_provider_addr=registry.terraform.io/hashicorp/aws tf_rpc=ApplyResourceChange @module=aws http.request.header.x_amz_content_sha256=e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855 http.request.header.x_amz_date=20230413T143208Z tf_resource_type=aws_s3_object timestamp=2023-04-13T15:32:08.799+0100
2023-04-13T15:32:08.846+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: HTTP Response Received: aws.sdk=aws-sdk-go aws.service=S3 http.response.header.accept_ranges=bytes tf_resource_type=aws_s3_object aws.region=eu-west-1 http.response.body= http.status_code=200 tf_provider_addr=registry.terraform.io/hashicorp/aws http.duration=46 http.response.header.x_amz_request_id=SW76TH7XATVCMSA9 http.response_content_length=94 tf_rpc=ApplyResourceChange http.response.header.server=AmazonS3 http.response.header.last_modified="Thu, 13 Apr 2023 14:32:09 GMT" http.response.header.x_amz_server_side_encryption=AES256 @caller=github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2@v2.0.0-beta.26/logger.go:138 aws.operation=HeadObject http.response.header.x_amz_id_2=xJ4W4evlwCXWK3Onxvm2p45S9hHH/pASpDuhXE+sd2+Axv6a83qcglXRBBQ00DL2lF3sk7m3IKQ= @module=aws http.response.header.content_type=binary/octet-stream http.response.header.etag="4b08369cef41be53c636bbf20cc11210" http.response.header.date="Thu, 13 Apr 2023 14:32:09 GMT" tf_mux_provider=*schema.GRPCProviderServer tf_req_id=09b7b2d1-5bdb-8486-6e08-f3852a8631b3 timestamp=2023-04-13T15:32:08.846+0100
2023-04-13T15:32:08.846+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] Waiting for state to become: [success]
2023-04-13T15:32:08.846+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: [DEBUG] Waiting for state to become: [success]
2023-04-13T15:32:08.847+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: HTTP Request Sent: http.request.header.x_amz_content_sha256=e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855 http.request.header.x_amz_security_token=***** http.user_agent="APN/1.0 HashiCorp/1.0 Terraform/1.4.5 (+https://www.terraform.io) terraform-provider-aws/4.62.0 (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go/1.44.237 (go1.19.7; linux; amd64)" net.peer.name=0c79c024-7ebb-d260-2f8b-8630b1dfc45d.s3.eu-west-1.amazonaws.com @module=aws aws.operation=GetObjectTagging aws.sdk=aws-sdk-go tf_provider_addr=registry.terraform.io/hashicorp/aws aws.region=eu-west-1 http.flavor=1.1 http.method=GET http.url=https://0c79c024-7ebb-d260-2f8b-8630b1dfc45d.s3.eu-west-1.amazonaws.com/manifest.json?tagging= tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2@v2.0.0-beta.26/logger.go:90 aws.service=S3 http.request.header.authorization="AWS4-HMAC-SHA256 Credential=ASIA************WFMH/20230413/eu-west-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date;x-amz-security-token, Signature=*****" tf_req_id=09b7b2d1-5bdb-8486-6e08-f3852a8631b3 tf_resource_type=aws_s3_object http.request.body= http.request.header.x_amz_date=20230413T143208Z tf_mux_provider=*schema.GRPCProviderServer timestamp=2023-04-13T15:32:08.846+0100
2023-04-13T15:32:08.892+0100 [DEBUG] provider.terraform-provider-aws_v4.62.0_x5: HTTP Response Received: http.status_code=200 tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2@v2.0.0-beta.26/logger.go:138 http.response.header.date="Thu, 13 Apr 2023 14:32:09 GMT" aws.operation=GetObjectTagging aws.sdk=aws-sdk-go http.duration=45 tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_id=09b7b2d1-5bdb-8486-6e08-f3852a8631b3 tf_resource_type=aws_s3_object aws.service=S3 http.response.body="<?xml version="1.0" encoding="UTF-8"?>
<Tagging xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><TagSet/></Tagging>" tf_mux_provider=*schema.GRPCProviderServer aws.region=eu-west-1 http.response.header.x_amz_id_2=sVki9O3tH9mnG6T796lEs/XcyhMyxYB4tw1KWTDomHbhSaciqkUnS97i2j7hewohZBntXH8YxYY= http.response.header.x_amz_request_id=SW77VJQ0S1KC4G9S @module=aws http.response.header.server=AmazonS3 timestamp=2023-04-13T15:32:08.892+0100
2023-04-13T15:32:08.892+0100 [WARN]  Provider "provider[\"registry.terraform.io/hashicorp/aws\"]" produced an unexpected new value for aws_s3_object.object, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .etag: was cty.StringVal("4926ff6720d81767034d1f8aa57e4dd8"), but now cty.StringVal("4b08369cef41be53c636bbf20cc11210")
aws_s3_object.object: Modifications complete after 0s [id=manifest.json]
╷
│ Error: error creating QuickSight Data Set: InvalidParameter: 1 validation error(s) found.
│ - missing required field, CreateDataSetInput.LogicalTableMap[id].DataTransforms[0].ProjectOperation.ProjectedColumns.
│ 
│ 
│   with aws_quicksight_data_set.test,
│   on main.tf line 52, in resource "aws_quicksight_data_set" "test":
│   52: resource "aws_quicksight_data_set" "test" {
│ 
╵
2023-04-13T15:32:08.922+0100 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-04-13T15:32:08.929+0100 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/aws/4.62.0/linux_amd64/terraform-provider-aws_v4.62.0_x5 pid=91361
2023-04-13T15:32:08.929+0100 [DEBUG] provider: plugin exited
```

### Panic Output

_No response_

### Important Factoids

_No response_

### References

https://github.com/hashicorp/terraform-provider-aws/pull/30349

### Would you like to implement a fix?

None

--- a/.changelog/30699.txt
+++ b/.changelog/30699.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_quicksight_data_set: Fix to properly expand `project_operation.projected_columns` field
+```

--- a/internal/service/quicksight/data_set.go
+++ b/internal/service/quicksight/data_set.go
@@ -1396,8 +1396,8 @@ func expandDataSetProjectOperation(tfList []interface{}) *quicksight.ProjectOper
 	}
 
 	projectOperation := &quicksight.ProjectOperation{}
-	if v, ok := tfMap["projected_columns"].([]string); ok && len(v) > 0 {
-		projectOperation.ProjectedColumns = aws.StringSlice(v)
+	if v, ok := tfMap["projected_columns"].([]interface{}); ok && len(v) > 0 {
+		projectOperation.ProjectedColumns = flex.ExpandStringList(v)
 	}
 
 	return projectOperation

